x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/aggs/frequentitemsets/mr/DelegatingCircuitBreakerServiceTests.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.aggs.frequentitemsets.mr;
+
+import org.elasticsearch.common.breaker.CircuitBreaker;
+import org.elasticsearch.common.breaker.PreallocatedCircuitBreakerService;
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.indices.breaker.HierarchyCircuitBreakerService;
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.List;
+
+import static org.elasticsearch.indices.breaker.HierarchyCircuitBreakerService.REQUEST_CIRCUIT_BREAKER_LIMIT_SETTING;
+import static org.elasticsearch.indices.breaker.HierarchyCircuitBreakerService.USE_REAL_MEMORY_USAGE_SETTING;
+
+public class DelegatingCircuitBreakerServiceTests extends ESTestCase {
+
+    public void testThreadedExecution() {
+
+        try (
+            HierarchyCircuitBreakerService topBreaker = new HierarchyCircuitBreakerService(
+                Settings.builder()
+                    .put(REQUEST_CIRCUIT_BREAKER_LIMIT_SETTING.getKey(), "10mb")
+                    // Disable the real memory checking because it causes other tests to interfere with this one.
+                    .put(USE_REAL_MEMORY_USAGE_SETTING.getKey(), false)
+                    .build(),
+                List.of(),
+                new ClusterSettings(Settings.EMPTY, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS)
+            )
+        ) {
+            PreallocatedCircuitBreakerService preallocated = new PreallocatedCircuitBreakerService(
+                topBreaker,
+                CircuitBreaker.REQUEST,
+                10_000,
+                "test"
+            );
+
+            CircuitBreaker breaker = preallocated.getBreaker(CircuitBreaker.REQUEST);
+
+            DelegatingCircuitBreakerService delegatingCircuitBreakerService = new DelegatingCircuitBreakerService(breaker, (bytes -> {
+                breaker.addEstimateBytesAndMaybeBreak(bytes, "test");
+            }));
+
+            Thread consumerThread = new Thread(() -> {
+                for (int i = 0; i < 100; ++i) {
+                    delegatingCircuitBreakerService.getBreaker("ignored").addEstimateBytesAndMaybeBreak(i % 2 == 0 ? 10 : -10, "ignored");
+                }
+            });
+
+            final Thread producerThread = new Thread(() -> {
+                delegatingCircuitBreakerService.disconnect();
+                preallocated.close();
+            });
+            consumerThread.start();
+            producerThread.start();
+            consumerThread.join();
+            producerThread.join();
```

Comment:
Is it enough if the threads just finish?
Or should there be some assertion in the end checking the value of estimate bytes?

Reply:
the purpose of the test is not hitting this line: https://github.com/elastic/elasticsearch/blob/4602570fda5a5d7d67f2bdb798767e12c515eb19/server/src/main/java/org/elasticsearch/common/breaker/PreallocatedCircuitBreakerService.java#L109

I was able to re-produce the CI issue with  `-Dtests.iters=1000` - but I don't want to check-in a loop.

Still, I should document ^this, will do.